recipes/libfabric/all/conanfile.py

```diff
@@ -1,99 +1,209 @@
-from conans import ConanFile, AutoToolsBuildEnvironment, tools
-from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.35.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os, fix_apple_shared_install_name
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import copy, get, rm, rmdir
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.53.0"
+
 
 class LibfabricConan(ConanFile):
     name = "libfabric"
-    description = "Open Fabric Interfaces"
-    topics = ("fabric", "communication", "framework", "service")
+    description = ("Libfabric, also known as Open Fabrics Interfaces (OFI), "
+                   "defines a communication API for high-performance parallel and distributed applications.")
+    license = ("BSD-2-Clause", "GPL-2.0-or-later")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "http://libfabric.org"
-    license = "BSD-2-Clause", "GPL-2.0-or-later"
+    topics = ("fabric", "communication", "framework", "service")
+
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
-    _providers = ['gni', 'psm', 'psm2', 'psm3', 'rxm', 'sockets', 'tcp', 'udp', 'usnic', 'verbs', 'bgq']
+    _providers = [
+        "cxi",
+        "dmabuf_peer_mem",
+        "efa",
+        "hook_debug",
+        "hook_hmem",
+        "mrail",
+        "opx",
+        "perf",
+        "profile",
+        "psm2",
+        "psm3",
+        "rxd",
+        "rxm",
+        "shm",
+        "sm2",
+        "sockets",
+        "tcp",
+        "trace",
+        "ucx",
+        "udp",
+        "usnic",
+        "verbs",
+        "xpmem",
+    ]
     options = {
-        **{ p: "ANY" for p in _providers },
-        **{
-            "shared": [True, False],
-            "fPIC": [True, False],
-            "with_libnl": "ANY",
-            "with_bgq_progress": [None, "auto", "manual"],
-            "with_bgq_mr": [None, "basic", "scalable"]
-        }
+        "shared": [True, False],
+        "fPIC": [True, False],
+        **{ p: ["yes", "no", "dl"] for p in _providers },
     }
     default_options = {
-        **{ p: "auto" for p in _providers },
-        **{
-            "shared": False,
-            "fPIC": True,
-            "with_libnl": None,
-            "with_bgq_progress": None,
-            "with_bgq_mr": None
-        }
+        "shared": False,
+        "fPIC": True,
+        "cxi": "no",  # library not available on CCI
+        "dmabuf_peer_mem": "yes",
+        "efa": "yes",
```

Comment:
I also dislike "auto" as it may result in not reproducible scenarios (we had bugs in CCI), but where in the upstream did you find those default values?

I searched in other package managers too and most of them don't manage those options. Only Gentoo and Ubunut I found doing some configuration:

- https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-block/libfabric/libfabric-1.11.2-r2.ebuild#n43
- https://git.launchpad.net/ubuntu/+source/libfabric/tree/debian/rules?h=ubuntu/jammy#n17

Reply:
I enabled all features that are not blocked by external dependencies. I agree that it should instead match the project defaults. It was nice to see that they pass in CI at least, though.